File: WebApplication2/wwwroot/js/Esg.js
var datatable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    datatable = $('#tbltable').DataTable({
        /*取得資料*/
        "ajax": {
            "url": "/Customer/Esg/GetAll"
        },
        /*columns通常會是陣列*/
        "columns": [
            /*須注意變數名稱要和Json格式的變數名稱一致*/
            { "data": "companyNumber", "width": "15%" },
            { "data": "companyName", "width": "15%" },
            { "data": "susESG", "width": "15%" },
            { "data": "msciESG", "width": "15%" },
            { "data": "ftseESG", "width": "15%" },
            { "data": "issESG", "width": "15%" },
            { "data": "sapESG", "width": "15%" },
            { "data": "twCompanyRank", "width": "15%" },
            { "data": "refi", "width": "15%" }
        ],
        language: {
            url: "https://cdn.datatables.net/plug-ins/1.11.3/i18n/zh_Hant.json"
        }
    });

}

File: BulkyBook.Models/Esg.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BulkyBook.Models
{
    public class Esg
    {
        public int Id { get; set; }
        public string CompanyNumber { get; set; }
        public string CompanyName { get; set; }
        public string SusESG { get; set; }
        public string MsciESG { get; set; }
        public string FtseESG { get; set; }
        public string IssESG { get; set; }
        public string SapESG { get; set; }
        public string TwCompanyRank { get; set; }
        public string Refi { get; set; }
    }
}
